app-common/src/lib.rs

```diff
@@ -0,0 +1,231 @@
+use clap::Args;
+use rustls::{ClientConfig, ServerConfig, SupportedProtocolVersion, pki_types::{CertificateDer, CertificateRevocationListDer, PrivateKeyDer, pem::PemObject}, server::WebPkiClientVerifier};
+use tracing::{debug, info};
+use std::{path::PathBuf, sync::Arc};
+use snafu::{Snafu, ResultExt};
+
+#[derive(Snafu, Debug)]
+pub enum TLSError {
+    #[snafu(display("IO error: {}", source))]
```

Comment:
It is poor practice to print a registered error source in the display. Applications using snafu would use the [report abstraction](https://docs.rs/snafu/latest/snafu/#reporting).